### PR TITLE
Add Available Phone Numbers API

### PR DESCRIPTION
**Pull request by kevinburke**
PR for #14
@kevinburke please take a look on this

### CODE REVIEW

File: available_phone_numbers.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	Mobile   *AvailablePhoneNumberBase
+	TollFree *AvailablePhoneNumberBase
+}
+
+// The subresources of the AvailablePhoneNumbers resource let you search for local, toll-free and
+// mobile phone numbers that are available for you to purchase.
+// See https://www.twilio.com/docs/api/rest/available-phone-numbers for details
+type AvailablePhoneNumber struct {
+	FriendlyName        string            `json:"friendly_name"`
+	PhoneNumber         string            `json:"phone_number"`
```
Comment (kevinburke) [inline available_phone_numbers.go]: This should be a PhoneNumber type

File: available_phone_numbers.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	Lata                string            `json:"lata"`
+	RateCenter          string            `json:"rate_center"`
+	Latitude            string            `json:"latitude"`
+	Longitude           string            `json:"longitude"`
+	Region              string            `json:"region"`
+	PostalCode          string            `json:"postal_code"`
+	IsoCountry          string            `json:"iso_country"`
```
Comment (kevinburke) [inline available_phone_numbers.go]: We should probably name this ISOCountry instead.

File: available_phone_numbers.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+type AvailablePhoneNumberSearchResult struct {
+	Uri     string                  `json:"uri"`
+	Numbers []*AvailablePhoneNumber `json:"available_phone_numbers"`
+}
+
+// https://www.twilio.com/docs/api/rest/available-phone-numbers#local
+// https://www.twilio.com/docs/api/rest/available-phone-numbers#toll-free
+// https://www.twilio.com/docs/api/rest/available-phone-numbers#mobile
+func (s *AvailablePhoneNumberBase) Get(ctx context.Context, isoCountry string, filters url.Values) (*AvailablePhoneNumberSearchResult, error) {
```
Comment (kevinburke) [inline available_phone_numbers.go]: Should this be `GetPage`, and return you an `AvailablePhoneNumberPage`? We return an array of results. In other places, `Get` returns a single resource.

Comment (mxpv) [reply available_phone_numbers.go]: According to [documentation](https://www.twilio.com/docs/api/rest/available-phone-numbers#local) they don't return any page information. Can be a bit confusing :)

Comment (kevinburke) [reply available_phone_numbers.go]: Agreed, but the type system should help, and we're also not allowing or returning any of the Iterators we have on other resources.

**Comment by mxpv** [comment]
Agree. Adding PR with fixes to this branch

**Comment by kevinburke** [comment]
Merged as e9ccdb831bd17f877932b735f154bc8657a26fc7. Thanks!

**Comment by kevinburke** [comment]
This change is available in version 0.61

**Pull request closed without merging**